### ISSUE TITLE
Plexdrive malfunctions with non-standard characters in filename

### ISSUE THREAD

**Issue by WestSmith** [dweidenfeld/plexdrive#300]
Recently spent a couple days tracking an issue with Plexdrive. Plexdrive would refuse to load a specific sub-directory in GDrive. Did all the standard troubleshooting I read up in the forums (uninstalling Plexdrive, making sure Google didn't change file format on upload, reauthorizing API, etc). None resolved issue.

Eventually, discovered that a file had used the character | in it's name (not to be confused with lower case L or upper case i). Removing it, and waiting for Plexdrive to update the cache, immediately solved the issue.

**Comment by disenchant** [comment]
I ran into the same issue with a Google Maps file containing a slash character in it's name like "`vacation-2019/2020`".

**Comment by felixbuenemann** [comment]
I'm not quite sure what to do about that, since the forward slash (/) is reserved as a directory separator in most Unix systems.

**Comment by felixbuenemann** [comment]
What probably could be done is to completely ignore file names with illegal characters and logging a warning. Any form of substitution scheme is prone to conflicting with existing files, so I don't think it is a good idea.

**Comment by disenchant** [comment]
@felixbuenemann I fully agree and think that your proposed solution should be sufficient as having such a file is rather unlikely but would completely block any further indexing. So ignoring it and adding a log entry is in my opinion an acceptable way of handling it. Replacing the `/` character with something else can also be done and e.g. macOS is doing it with a `:` but as mentioned by you it could lead to file conflicts so I would rather not risk it for an edge case like this.

**Comment by WestSmith** [comment]
Thanks for the attention on this issue. For context: I didn't intend to name any files with "|" in the name. If I remember correctly, it was a pre-existing file I had downloaded. I don't normally use that in my file names when naming a file myself and I don't use any apps which use that as a naming convention. The solution proposed by @disenchant would be the ideal for someone of my (relatively limited) skill-level.
